generic-ec/src/arithmetic.rs

```diff
@@ -337,30 +339,46 @@ impl_binary_ops! {
     Mul (Scalar<E>, mul, Generator<E> = Point<E>) laws::mul_of_scalar_at_generator_is_valid_point,
 }
 
-// Point <> SecretScalar, Scalar <> SecretScalar, NonZero<Point> <> SecretScalar,
-// NonZero<Scalar> <> SecretScalar arithmetic ops
+// SecretScalar and NonZero<SecretScalar> multiplication, addition, substruction with Scalar,
+// NonZero<Scalar>, Point, and NonZero<Point>
 impl_binary_ops! {
```

Comment:
Damn these are hard to follow; and also they are a combinatorial explosion. In the future I want to format them as a table somehow

Reply:
Yeah, it's hard to read for me too. I'm usually looking at this expression that helps me to make sure that everything is implemented:

https://github.com/dfns/generic-ec/blob/c0ec6d016f95db5844fd4df96399c73b98ea15a0/generic-ec/src/arithmetic.rs#L622-L655